src/App.tsx

```diff
@@ -1,14 +1,80 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Select } from './types/Select';
+
+function hasNormalizedQuery(content: string, query: string) {
+  const normalizedQuery = query.trim().toLowerCase();
+
+  return content.toLowerCase().includes(normalizedQuery);
+}
+
+function filterBySelect(todo: Todo, selectedOption: string) : boolean {
+  if (!selectedOption || selectedOption === Select.All) {
+    return true;
+  }
+
+  if (selectedOption === Select.Active) {
+    return !todo.completed;
+  }
+
+  return todo.completed;
+}
+
+function filterTodosByQuery(todos: Todo[], query: string, selectedOption: string) {
+  if (!query && !selectedOption) {
+    return todos;
+  }
+
+  return todos.filter((todo) => (
+    hasNormalizedQuery(todo.title, query)
+    && filterBySelect(todo, selectedOption)
+  ));
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [showModal, setShowModal] = useState(false);
+  const [query, setQuery] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [showLoader, setShowLoader] = useState(true);
```

Comment:
```suggestion
  const [isLoading, setIsLoading] = useState(false);
```

Better to make `false` as a default value, and change it to `true` only before you are making API call